pkg/ottl/parser.go

```diff
@@ -480,6 +481,19 @@ func (p *Parser[K]) ParseValueExpression(raw string) (*ValueExpression[K], error
 	}
 
 	return &ValueExpression[K]{
-		getter: getter,
+		getter: &StandardGetSetter[K]{
```

Comment:
If we end up stick to this approach, considering the `mapGetter` is now returning `pcommon.Map` values, I think we could keep it consistent and remove these changes, so all maps would still be parsed as`pcommon.Map` instead of raw values. I guess it would be a breaking change, though. WDYT?

Reply:
Judging by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37280#discussion_r1925483223, we probably need to survey this a bit more to make sure we're either consistent everywhere, or can translate at any given point in the chain. I'm not sure I have a strong opinion here, but it sounds like at a minimum we could potentially use more tests.

Do the same issues apply to slices with `[]any` vs. `pcommon.Slice`?

Reply:
for slices, this issue doesn't apply, as the `listGetter` returns `[]any` as opposed to `pcommon.Slice`:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ee2d1650b814fb3e2f733baa969b1d495ad78c8d/pkg/ottl/expression.go#L164-L176

Reply:
I agree with @edmocosta that we should continue returning pcommon.Map

Reply:
Alright, I now changed it again to return pcommon.Map